modules/migration-about-migrating-applications-api.adoc

```diff
@@ -0,0 +1,112 @@
+// * migration-toolkit-for-containers/advanced-migration-options-mtc.adoc
+
+[id="migration-about-migrating-applications-api_{context}"]
+= About migrating applications
+
+You can migrate applications from a local cluster to a remote cluster, from a remote cluster to a local cluster, and between remote clusters.
+
+[id="terminology_{context}"]
+== Terminology
+
+Source cluster::
+* Cluster from which the applications are migrated.
+
+Destination cluster::
+* Cluster to which the applications are migrated.
+
+Replication repository::
+* Object storage.
+* Requires network access to all clusters.
+
+Indirect migration::
+* Images, volumes, and Kubernetes objects are copied from the source cluster to the replication repository and then from the replication repository to the destination cluster.
+
+Direct volume migration::
+* Volumes are copied directly from the source cluster to the destination cluster.
+* Significantly faster than indirect migration.
+
+Direct image migration::
+* Images are copied directly from the source cluster to the destination cluster.
+* Significantly faster than indirect migration.
+
+Host cluster::
+* Cluster on which the `migration-controller` pod and the web console run.
+* Usually the same as the destination cluster and the local cluster but this is not a requirement.
+* Does not require an exposed secure registry route for direct image migration.
+
+Local cluster::
+* Usually the same as the destination cluster and the host cluster but this is not a requirement.
+
+Remote cluster::
+* Usually the same as the source cluster but this is not a requirement.
+* Requires an exposed secure registry route for direct image migration.
+* Requires a `Secret` CR containing the `migration-controller` service account token.
+
+[id="mapping-destination-namespaces-in-the-migplan-custom-resource_{context}"]
+== Mapping destination namespaces in the MigPlan custom resource (CR)
+
+If you map destination namespaces in the `MigPlan` CR, you must ensure that the namespaces are not duplicated on the source or the destination clusters because the UID and GID ranges of the namespaces are copied during migration.
+
+The following examples will cause validation errors:
+
+.Two source namespaces mapped to the same destination namespace
+[source,yaml]
+----
+spec:
+  namespaces:
+    - namespace_2
+    - namespace_1:namespace_2
+----
+
+.Source namespace mapped to duplicate namespace
+[source,yaml]
+----
+spec:
+  namespaces:
+    - namespace_1:namespace_1
+----
+
+[id="using-migmigration-crs-for-stage-migration-cancellation-and-rollback_{context}"]
+== Stage migration, migration cancellation, and migration rollback
+
+You can create and associate multiple `MigMigration` custom resources (CRs) with the same `MigPlan` CR for the following use cases:
+
+* To perform stage migrations, which copy most of the data without stopping the application, before running a migration. Stage migrations improve the performance of the migration.
```

Comment:
I think this sentence captures the spirit of what stage does, although technically stage copies _all data_ available at the time of the stage.

Reply:
I'm also not sure that `Stage migrations improve the performance of the migration.` is painting the full picture. 

Running a stage migration first will reduce the cutover window (downtime of apps being migrated), but the total migration time will be extended at least a little bit since you are now running two migrations, and the final migration still have to compare all files on PVs for delta copying.

Here's an illustration of the difference (high level ideas only, not including all phases)
![image](https://user-images.githubusercontent.com/7576968/122998109-a3a67c80-d37a-11eb-8bc1-eea5248f9d95.png)